source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -349,6 +349,11 @@ export abstract class CardElement extends CardObject {
         }
     }
 
+    protected updateInputsVisualState(hover: boolean) {
```

Comment:
Is there a reason why this is added at the `CardElement` level instead of at the `AdaptiveCard` or `Container` level?

It looks like this method is only called from within `AdaptiveCard`

Reply:
David has suggested this. I tried to put this in AdaptiveCard class but this.getAllInputs() was not available there. So Looks like AdaptiveCard does not implement getAllInputs

Reply:
I like your approach here that updateInputVisualState is called once per visual update, but it is a bit confusing that the call is implemented at the CardElement. Can you change the name of this api to more distinct from UpdateVisualState() and add a comment that this is mean to be called only at the card level?

Reply:
Do you have any suggestions in mind? This "updateInputsVisualState" was suggested by David and I see similar methods names in CardElement as "updateActionsEnabledState", 
![image](https://user-images.githubusercontent.com/12997454/214239525-46e5670c-ba53-4661-94e4-dcd75f2dd12b.png)

Reply:
I have added comment on this class indicating why input.updateVisualState() needs to be called on card level on mouse events.

Also input.updateVisualState() is called from other places as well like onBlur event on input or during initial render of inputs. So this updateVisualState is not exclusive to card level events.

Reply:
I meant only for updateInputVisualState. It's really confusing when the function names are different just a little bit. updateInputsVisualState vs updateVisualState